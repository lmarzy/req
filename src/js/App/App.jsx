import React, { Component } from 'react';
import { Header, Footer } from 'components';
import helpers from '../helpers';

import '../../css/app.scss';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      siteName: 'REQ',
      tagLine: 'Crafting the web',
      year: helpers(),
    };
  }

  render() {
    const { siteName, tagLine, year } = this.state;
    return (
      <div>
        <Header siteName={siteName} tagLine={tagLine} />
        <main>
          <section>
            <h2>About</h2>
          </section>
        </main>
        <Footer year={year} />
      </div>
    );
  }
}

export default App;
