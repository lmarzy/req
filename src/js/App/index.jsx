import React, { Component } from 'react';
import { Header, Footer } from 'components';
import helpers from '../helpers';

import '../../css/app.scss';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      siteName: 'REQ',
      year: helpers(),
    };
  }

  render() {
    return (
      <div>
        <Header siteName={this.state.siteName} />
        <main>
          <section>
            <h2>About</h2>
          </section>
        </main>
        <Footer year={this.state.year} />
      </div>
    );
  }
}

export default App;
