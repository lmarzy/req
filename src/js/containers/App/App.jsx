import React, { Component } from 'react';
import { Header, Footer } from 'components';
import { Home } from 'containers';

import '../../../css/app.scss';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      siteName: 'Req',
      tagLine: 'Crafting a better web',
    };
  }

  render() {
    const { siteName, tagLine } = this.state;

    return (
      <div>
        <Header siteName={siteName} tagLine={tagLine} />
        <main>
          <Home />
        </main>
        <Footer />
      </div>
    );
  }
}

export default App;
