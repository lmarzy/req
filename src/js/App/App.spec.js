import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import App from './App';

const component = shallow(<App />);
const tree = renderer.create(<App />).toJSON();

describe('<App />', () => {
  it('Should render 1 <App /> component', () => {
    expect(component).toHaveLength(1);
  });

  it('Should render without crashing', () => {
    const div = document.createElement('div');
    render(<App />, div);
  });

  it('Render a snapshot', () => {
    expect(tree).toMatchSnapshot();
  });
});
