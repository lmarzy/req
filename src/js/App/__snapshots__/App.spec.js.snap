// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Render a snapshot 1`] = `
<div>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <h1
        className="logo"
      >
        REQ
      </h1>
      <p
        className="tagline"
      >
        Crafting the web
      </p>
    </div>
  </header>
  <main>
    <section>
      <h2>
        About
      </h2>
    </section>
  </main>
  <footer>
    <small>
      Â© 
      2018
    </small>
  </footer>
</div>
`;
